### PR TITLE
Signal Processing Improvements and Configuration Name Changes

### PR DESCRIPTION
**Pull request by jganoff**
I'm hoping to avoid the overhead of splitting this PR into two. These changes modify the image's behavior as follows:

1. Signals sent from Docker are now received by supervisord and properly sent to the start-flink.sh script and consequently the Flink daemon. This means the image properly responds to `docker stop`.
2. Supervisord is killed, and therefore the container, if the Flink daemon fails to start or dies. This may not be desirable when running in Docker standalone but within a Swarm cluster or Kubernetes it's preferable to let the orchestration layer handle restarts rather than the container itself.
3. Environment variables updated to more closely resemble the Flink configuration they overwrite. I also exposed more configuration that can be overridden.
4. I redirected all console logging from the start-flink.sh scripts to stdout so they show up in `docker logs`.
5. Flink process log and job output (flink-*.log and flink-*.out) are now redirected to stdout as well. This also improves the robustness of the tail command and waits for the files to exist before attempting to tail them - removing any race condition during start up.

### CODE REVIEW

**Comment by mjburghoffer** [comment]
@jganoff Thanks for submitting this PR!  We are going to merge it in now.  Great work!

File: scripts/start-flink.sh [hunk 1]
```diff
@@ -22,47 +29,54 @@ function edit_properties() {
...
+while [ ! -f $FLINK_HOME/log/flink-*.log ]; do
+  sleep 1
+done
+
+# Pipe Flink output to stdout
+tail -10000 -f $FLINK_HOME/log/flink-*.log &
+tail -10000 -f $FLINK_HOME/log/flink-*.out &
+
+# Wait for Flink process to terminate...
+PID=$(cat $PID_FILE)
```
Comment (nari-m) [inline scripts/start-flink.sh]: will this ever get executed since you have tail -f as a previous command ?

Comment (jontg) [inline scripts/start-flink.sh]: @nari-m yes — he forks the process with `&`

**Comment by jganoff** [comment]
Thanks for merging!

**Pull request merged**